### PR TITLE
Stream zip file from s3 to the client

### PR DESCRIPTION
**Pull request by aroltsch**
connects #475 

We need to test it on UAT.

Note: I will update Commons  once the commons PR is merged.

### CODE REVIEW

File: app/services/download_documents.rb [hunk 1]
```diff
@@ -115,11 +115,9 @@ def zip_exists_locally?
     File.exist?(zip_path)
   end
 
-  def fetch_zip_from_s3
-    # if the file exists on the filesystem, skip
-    return if zip_exists_locally?
-
-    @s3.fetch_file(@download.s3_filename, zip_path)
+  def stream_zip_from_s3
+    key = Rails.application.config.s3_enabled ? @download.s3_filename : zip_path
```
Comment (shanear) [inline app/services/download_documents.rb]: this could be moved to a function `def streaming_s3_key`

Comment (aroltsch) [inline app/services/download_documents.rb]: good idea

**Comment by aroltsch** [comment]
We will be testing it using this branch so I will not merge it yet.

**Comment by askldjd** [comment]
Tested pretty extensively for an hour. 

Verified:
 - Download streams immediately
 - Zero files are in tmp folder
 - Little to zero CPU usage

Side note: I also found out that VA network routers is probably buffering our transfer. I started 4 500MB download, killed all EC2 within 1 min of download, and all files were transferred successfully. Our ELB connection draining parameters is probably not doing much because of this.

**Pull request merged**